openedx/openedx-wordpress-ecommerce
# fix: remove the exclusion of NonceVerification in phpcs and fix the issues

## Description

This PR:
- Adds [nonces](https://developer.wordpress.org/apis/security/nonces/) in the forms of our plugin (settings, enrollment, product and order).
- Removes the exclusion of the **WordPress.Security.NonceVerification.Missing**  alert.
- Changes the action `save_post_shop_order` to `woocommerce_update_order`.

## Testing instructions

- **Check the tests are passing:** To know if we are following the WordPress standards without avoiding alerts.
- **Check if the plugin works as expected in WordPress:** Principally the workflows
    - Save and update the plugin settings ([Reference image](https://github.com/openedx/openedx-wordpress-ecommerce/blob/main/docs/source/_images/how-tos/create_an_openedx_app/openedx-sync-plugin-settings.png))
    - Create or edit a manual enrollment. ([Reference](https://github.com/openedx/openedx-wordpress-ecommerce/blob/main/docs/source/how-tos/create_enrollment_requests_manually.rst))
    - Create or edit an Open edX course product ([Reference](https://github.com/openedx/openedx-wordpress-ecommerce/blob/main/docs/source/how-tos/create_openedx_course_wordpress.rst))
    - Create or edit a WooCommerce Order with an Open edX course product and change the Enrollment associated.

The expected result is that the forms are saved correctly (if the nonce_verification fails, the save or update won't store the new information).

Credentials for the stage to test: https://share.1password.com/s#Pu505pb8w5wyFnHuHhvHBg4K1B5VIHk69kK0O-mCy9s

## Additional information

- This change favors publishing this plugin in the WordPress org and following the WordPress PHPCS standards.
- I changed `save_post_shop_order` to `woocommerce_update_order` because the first one was preventing me from saving the order extra information. 

**More references related:**

- [Nonces](https://developer.wordpress.org/apis/security/nonces/)
- [Understand and use WordPress Nonces](https://developer.wordpress.org/news/2023/08/01/understand-and-use-wordpress-nonces-properly/)
- [Hooks WooCommerce](https://woocommerce.github.io/code-reference/hooks/hooks.html)
- [woocommerce_update_order](https://woocommerce.github.io/code-reference/files/woocommerce-includes-data-stores-class-wc-order-data-store-cpt.html#source-view.196)

## Checklist for Merge

- [x] Tested in a remote environment (I tested all the workflows mentioned and the full shopping workflow)
